Split user registration in two steps

ferblape opened the pull request:
#### :tophat: What? Why?

See https://meta.decidim.org/processes/roadmap/f/122/proposals/14855

Context: We’d like to improve the registration process for new users. The actual process is quite long and difficult, and we think Decidim should be more accessible and appealing.

This PR splits the signup process in two steps, and adds an optional steps for users to complete their profile as their first action in Decidim.

#### :pushpin: Related Issues

- Related to https://github.com/OpenSourcePolitics/dev/issues/11

#### :clipboard: Subtasks
- [x] Add `CHANGELOG` entry
- [x] Add tests

### :camera: Screenshots (optional)

**step 1**

![Screen Shot 2019-12-09 at 13 28 56](https://user-images.githubusercontent.com/17616/70435805-1b0e0700-1a88-11ea-9c2f-2db15fd7d620.png)

**step 2**

![Screen Shot 2019-12-09 at 18 25 07](https://user-images.githubusercontent.com/17616/70457868-40633b00-1ab1-11ea-91e1-6f2b21f03890.png)

paulinebessoles commented:
Hi ! We tested the two steps sign up registration and have a few feedbacks : 
### Registration steps
- [ ] the 2 steps are not visible for the user, they should be displayed on each step : 

**Step 1 :** 
<img width="486" alt="Capture d’écran 2019-12-12 à 11 38 37" src="https://user-images.githubusercontent.com/52420208/70705262-045bf000-1cd4-11ea-999e-62e3386c2706.png">
**Step 2 :** 
<img width="476" alt="Capture d’écran 2019-12-12 à 11 39 48" src="https://user-images.githubusercontent.com/52420208/70705311-1e95ce00-1cd4-11ea-9076-5009bdd14b71.png">

- [ ] There is twice the same button (send again confirmation information) on the last card of the steps : 
<img width="610" alt="Capture d’écran 2019-12-12 à 11 40 32" src="https://user-images.githubusercontent.com/52420208/70706420-25254500-1cd6-11ea-8e5a-32c341e7e835.png">

- [ ] When there is an error on the registration form (password not matching at step 1), I still can go on step 2. This should'nt be possible. A check (verification) of the information filled in should be made at each step.

- [ ] Furthermore, when there is an error, the two steps appear on the same screen. 
<img width="632" alt="Capture d’écran 2019-12-12 à 11 43 15" src="https://user-images.githubusercontent.com/52420208/70705727-d6c37680-1cd4-11ea-9fae-528188ea000f.png">

- [ ] As a consequence, when there is an error, the three cta buttons are displayed at the end of the form : 
<img width="611" alt="Capture d’écran 2019-12-12 à 11 48 46" src="https://user-images.githubusercontent.com/52420208/70706032-6bc66f80-1cd5-11ea-929f-b3b98205007d.png">
In this case, when clicking on continue, I'm redirected on an empty form : 
<img width="708" alt="Capture d’écran 2019-12-12 à 11 52 46" src="https://user-images.githubusercontent.com/52420208/70706328-fc04b480-1cd5-11ea-90c0-899a164d9178.png">

### Onboard screen

- [ ] There is no margin below the avatar image : 
<img width="1284" alt="Capture d’écran 2019-12-12 à 11 56 20" src="https://user-images.githubusercontent.com/52420208/70706598-7a615680-1cd6-11ea-903f-4bfcad0cd0d9.png">

ferblape commented:
Hi @paulinebessoles this is ready to review again, tests are looking good.

We have implemented all the feedback you provided and fixed the issue with the buttons.

carolromero commented:
Hi @ferblape do you have a link where we can check this? Thanks!

paulinebessoles commented:
Hi @ferblape,
Thanks for the modifications! 
I just have one more feedback : could you change the scroll after the first step ? 
After confirming the first step parameters, the user should be sent at the top of the page to start filling the second step (see below).
![Screenshot_2019-12-17 S'inscrire - Pink - DEV(1)](https://user-images.githubusercontent.com/52420208/71019329-754b3f80-20fa-11ea-8745-69575687a0de.png)

ferblape commented:
@paulinebessoles ready!

virgile-dev commented:
Hey @carolromero you check this out on : 
https://pink.osp.cat/users/sign_up (there are a couple customization but you'll get the idea). Do you think it can be merged soon ?

paulinebessoles commented:
Hi @ferblape I just have one issue before merge about the mobile display of the 2 steps registration : 
The red circle of the first step is not correctly aligned with the horizontal step 

![Capture d’écran - 2020-02-19 à 18 11 51](https://user-images.githubusercontent.com/52420208/74857108-7f302f80-5343-11ea-811b-4054031b907c.png)

@carolromero Do you think about something else ?

furilo commented:
@Crashillo can you fix this please?

Crashillo commented:
We're trying to merge this branch, but we're noticing all gitflow has been changed, so I'm afraid this branch could be outdated.
Look at the current flow we're doing: a PR from a forked repository.

I solved such conflicts merging master in the forked repository; but here they're still present. Could you please, help us? cc/ @mrcasals @tramuntanal

carolromero commented:
> @carolromero Do you think about something else ?

Hi @paulinebessoles, yes, when we checked this, we wrote down a couple of things:
1. The checkboxes, both for the conditions of use, and for receiving the newsletter, are in the first step, but could go in the second step better, no?
2. While this was being developed we added an improvement for the username field (https://github.com/decidim/decidim/pull/5495), which caused confusion, and that could be added to this form.
![imatge](https://user-images.githubusercontent.com/23509895/74927159-6709f000-53d7-11ea-8c10-fb4663d3183a.png)

3. Finally, the third step, where interests are defined, we see that it has limitations due to its design. For example, in the real case of Barcelona, there are more than 90 areas of interest. Regarding this, either:
  - We redesign the interests page, which is something that is best done independently so that it can be scaled.
![imatge](https://user-images.githubusercontent.com/23509895/74924969-b3ebc780-53d3-11ea-81f2-612894e5edcc.png)
  - We modify the third step in the registration form, and at the end, instead of showing the whole list of interests, we can show some links like:
     - Complete your profile
     - Define your interests
     - Start participating

@furilo @Crashillo as we commented offline with @virgile-dev, this PR shouldn't enter to master right now, because we are in the middle of a huge process in Barcelona and we cannot change the form at this time. What we agreed is that we would wait until June to merge. In the meantime we can consider these changes if it's okay with you.  In any case, we are not going to make any changes to the registration during the next months, so we minimize the number of changes that need to be made when the PR goes to master.

Leusev commented on CHANGELOG.md:
```diff
@@ -14,6 +14,17 @@ Now is its possible to configure every organization (tenant) with a different ti
 
 To upgrade it is recommended to configured the proper time zone in the admin for the organization and remove any `config.time_zone` personalization in Rails (unless you know what you are doing).
 
+- **many modules**: Added all spaces and many entities to global search, see Upgrade notes for more detail. [\#5469](https://github.com/decidim/decidim/pull/5469)
```
Review this lines, it seems there's been a bad merge.

tramuntanal commented:
As @product stated at the bottom of [its last comment](https://github.com/decidim/decidim/pull/5554#issuecomment-588964396) we would wait until June to merge (or when the PAM finishes).
Also tests, should be reviewed before merging.

virgile-dev commented:
@carolromero Now that PAM is over could re-open and see if we can merge this ?
@Crashillo do you think you could make the test the failing test pass on this ? 
Thanks everyone in advance.

virgile-dev commented:
@andreslucena @carolromero do you think we could merge this now ?

carolromero commented:
@virgile-dev what's the status of this? Were the changes we discussed [here](https://github.com/decidim/decidim/pull/5554#issuecomment-588964396) applied?

paulinebessoles commented:
Hi @carolromero ! 

We're going to work again on this one to be able to merge it. 

1. Okay to move the TOS agreement to step 2. The Newsletter checkbox has already been moved to the end of step 2. 
2. Definitely! I just tested locally and this is already implemented. 
3. It would be awesome to refactor the Interest page, but we don't have that much resources at the moment. I think the best option right now is to go with the 2nd solution. 
I think we should keep the "Complete your profile" section so users can modify their informations directly, but I like the idea of different links to complete user settings. 
In the following mockup, I added 3 links to set up notifications, authorizations and interests: 

![Frame 5](https://user-images.githubusercontent.com/52420208/91182836-fbd50980-e6ea-11ea-8783-bda3c0067342.png)
